src/domain/balances/balances.repository.ts

```diff
@@ -26,7 +32,26 @@ export class BalancesRepository implements IBalancesRepository {
       trusted: args.trusted,
       excludeSpam: args.excludeSpam,
     });
-    return balances.map((balance) => this.validator.validate(balance));
+    return balances.map((balance) => this.balancesValidator.validate(balance));
+  }
+
+  async getSimpleBalances(args: {
```

Comment:
Adding new functionality to `IBalancesRepository` for the migration does not seem to be the best way forward.

From the domain perspective, the `BalancesRepository` was already returning balances and fiat prices. Therefore I would keep the same assumption on the `domain` layer resulting in no changes in the `routes` layer.

What do you think?

Reply:
Not sure I follow, do you mean returning balances + prices (`Balance` entities) from `BalancesRepository` instead of having these two functions (one returning `Balance` entities and the other returning `SimpleBalance` entities)?

Reply:
Exactly 👍 The routes should not change because we get the pricing info from a different provider now. For the routes the `Balance` interface should still be followed.

Reply:
I agree, but that would imply calculating ' fiatBalance` and `fiatConversion` into the repository, right? So we would need to inject `PriceRepository` into `BalancesRepository`, or remove `PriceRepository` and integrate its functions into `BalancesRepository`, therefore calling `CoinegeckoApi` from there. 

I'm not against it. In fact, I think it makes perfect sense, I'm just aligning expectations.

Reply:
Yes, totally get your point. I was structuring this more like a migration (this feature is behind a FF), so in the end the routes would expect to have balances without fiat, and keep doing calculations on the routes layer.

But I agree it makes more sense to move these calculations to the domain layer (to the repository in this case), and by doing so, we can also abstract the routes from the balances composition, so we would benefit from that also. I will change this, thanks!

Reply:
I've changed the approach in [17040f3](https://github.com/safe-global/safe-client-gateway/pull/732/commits/17040f3f07ad63a3682ce6b665dc451dfae545a2) based on your suggestion.

Now the domain layer only exposes a function to get `Balances`:
```javascript
  getBalances(args: {
    chainId: string;
    safeAddress: string;
    trusted?: boolean;
    excludeSpam?: boolean;
  }): Promise<Balance[]>;
``` 
Some _ code pseudo-duplication_ in the routes layer service `BalancesService` (between `getBalances/_getFromSimpleBalances` and `mapBalance/_mapSimpleBalance`) still exists. I tried to not move completely the existing implementation to `BalancesRepository` (bringing the `IExchangeRepository` injection there and so on). This is intended to simplify the PR. I'm also considering this state (having both Coingecko and the Exchange API) should be transitory. 

If you think we should move the Exchange API to the repository also, to simplify the routes layer further (even if this should be a transitory state), I'd suggest we do it separately in a different PR.